text/0035-physics.md

```diff
@@ -0,0 +1,175 @@
+- Clients will receive position updates
+
+### Mixed Simulation Player Collision
+
+If two player controlled dynamic rigidbodies collide this will result in two dynamic/kinematic collisions as if both players where running against an immovable object.
+
+### Mixed Simulation Joints
+
+Joints between two dynamic rigidbodies which are owned by different peers will not resolve properly. This is not a scenario which we can support. The joints will instead act as if all the the Rigidbodies which are not controlled by the controller of the corresponding Rigidbody were kinematic.
+
+### Mixed Simulation Forces
+
+Forces can only be applied to a NetworkRigidbody by the peer which is controlling it. Often forces are applied in a specific area, by applying these forces on every peer all the NetworkRigidbodies will be affected correctly.
+
+### Interaction With Non Networked Physics Objects
+
+NetworkRigidbodies can interact and collide with regular Rigidbodies. Since this is often unwanted behavior it is recommended to either use a different collision layer for non networked rigidbodies or to move them into a separate physics scene entirely.
+
+## Why we do not synchronize velocity
+
+There are two reasons why we would need to synchronize velocity.
+1. Prediction: velocity allows for a more accurate prediction.
+2. Extrapolation in case of packet loss.
+
+We do not support prediction currently so 1. is not relevant. Regarding extrapolation while this might improve the smoothness of the Rigidbody when extrapolating in case of a lag due to packet loss it is non trivial to implement this without resulting in jitter. What we would need to do here is transition the Rigidbody from a simply replicating mode into a dynamic Rigidbody again and run the physics simulation to extrapolate. We currently do not have a good way to make this transition smooth because we are not in control of the FixedUpdate and the physics simulation. Instead simply using the NetworkTransforms extrapolation mode results in a more smooth experience and keeps our implementation simple.
+
+## Why we do not synchronize triggers and collision events
+
+1. Collision and Trigger events contain information in reference types (classes) which is not something we could properly serialize. In addition these events contain information about the components and GameObject involved in the collision. We currently have no way to serialize that information over the network.
+
+2. Assuming we could serialize the data of the collision event there's an other issue. When sending collisions we could either send them reliable or unreliable both of them have downsides:
+    1. Unreliable: Would kind of defeat the purpose because collisions and triggers would still not be fully reliable
+    2. Reliable: If the packet gets dropped the collision even will be delayed compared to the position sync.
```

Comment:
what would be the result? We'd see a ball bump (or miss the wall because of interpolation) on a wall, then 1+ RTT later we'd see the trigger event? Would it be that bad?

Reply:
I think delayed triggers/collisions is something we could live with. But it's just one more thing which makes syncing collision events iffy.

Since there are so many nuances and different ways to do it right I'd rather tell people to implement a simplified version which fits their need instead of trying to provide a one size fits all solution.